jetty-http2/http2-common/src/main/java/org/eclipse/jetty/http2/api/Stream.java

```diff
@@ -227,8 +227,24 @@ default boolean onIdleTimeout(Stream stream, Throwable x)
          * @param stream the stream
          * @param error the error code
          * @param reason the error reason, or null
+         * @param failure the failure
          * @param callback the callback to complete when the failure has been handled
          */
+        default void onFailure(Stream stream, int error, String reason, Throwable failure, Callback callback)
```

Comment:
I find this new method's signature and default implementation surprising. I would have expected that the `error` and `reason` parameters to go away especially since the overriding implementations just ignore them. Plus the fact that the default implementation drops the `failure` in favor of calling the deprecated implementation instead of directly succeeding the callback also makes me scratch my head. 

Maybe `FailureFrame.getFailure()` should return some sort of `H2Exception` that wraps the original exception as well as containing the reason and error code?

Reply:
This is at the HTTP/2 low-level API so our implementation may not use `error` and `reason` but users of these APIs may.
However, there are cases where we still have a failure but not an associated `error` (e.g. idle timeout and aborts).